runtime/runtime_impl_jetty9/src/main/java/com/google/apphosting/runtime/jetty9/JettyRequestAPIData.java

```diff
@@ -329,6 +327,41 @@ public String getScheme() {
           public boolean isSecure() {
             return isSecure;
           }
+
+          @Override
+          public String getRemoteAddr() {
+            return finalUserIp;
+          }
+
+          @Override
+          public String getServerName() {
+            return "0.0.0.0";
+          }
+
+          @Override
+          public String getRemoteHost() {
+            return finalUserIp;
+          }
+
+          @Override
+          public int getRemotePort() {
+            return 0;
+          }
+
+          @Override
+          public String getLocalName() {
+            return "0.0.0.0";
+          }
+
+          @Override
+          public String getLocalAddr() {
+            return "0.0.0.0";
+          }
```

Comment:
[nit] Would prefer to have these as String constants since they are repeated (similar to `WARMUP_IP)

Reply:
done